src/main/java/tp/cap5buddy/contacts/Email.java

```diff
@@ -0,0 +1,49 @@
+package tp.cap5buddy.contacts;
+
+import static java.util.Objects.requireNonNull;
+
+public class Email {
+
+    private static final String SPECIAL_CHARACTERS = "!#$%&'*+/=?`{|}~^.-";
+    public static final String MESSAGE_CONSTRAINTS = "Emails should be of the format local-part@domain "
+            + "and adhere to the following constraints:\n"
+            + "1. The local-part should only contain alphanumeric characters and these special characters, excluding "
+            + "the parentheses, (" + SPECIAL_CHARACTERS + ") .\n"
+            + "2. This is followed by a '@' and then a domain name. "
+            + "The domain name must:\n"
+            + "    - be at least 2 characters long\n"
+            + "    - start and end with alphanumeric characters\n"
+            + "    - consist of alphanumeric characters, a period or a hyphen for the characters in between, if any.";
+    // alphanumeric and special characters
+    private static final String LOCAL_PART_REGEX = "^[\\w" + SPECIAL_CHARACTERS + "]+";
+    private static final String DOMAIN_FIRST_CHARACTER_REGEX = "[^\\W_]"; // alphanumeric characters except underscore
+    private static final String DOMAIN_MIDDLE_REGEX = "[a-zA-Z0-9.-]*"; // alphanumeric, period and hyphen
+    private static final String DOMAIN_LAST_CHARACTER_REGEX = "[^\\W_]$";
+    public static final String VALIDATION_REGEX = LOCAL_PART_REGEX + "@"
+            + DOMAIN_FIRST_CHARACTER_REGEX + DOMAIN_MIDDLE_REGEX + DOMAIN_LAST_CHARACTER_REGEX;
```

Comment:
Might want to pull this out into a new class for common usage. As I foresee we will be using this for some of the other classes as well.

Reply:
This will be updated in a future PR.